<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Rashid Lab Website </title> <meta name="author" content="Naim Rashid"> <meta name="description" content="#A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta name="keywords" content="Biostatistics, Cancer, AI, Machine Learning, Clinical Trials"> <meta property="og:site_name" content="Rashid Lab Website"> <meta property="og:type" content="website"> <meta property="og:title" content="Rashid Lab Website | about"> <meta property="og:url" content="https://naimurashid.github.io/"> <meta property="og:description" content="#A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta property="og:image" content="/assets/img/RashidLabMainLogo2400x1800.jpeg"> <meta property="og:locale" content="en"> <meta name="twitter:card" content="summary"> <meta name="twitter:title" content="about"> <meta name="twitter:description" content="#A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta name="twitter:image" content="/assets/img/RashidLabMainLogo2400x1800.jpeg"> <meta name="twitter:site" content="@naimurashid# your X handle"> <meta name="twitter:creator" content="@naimurashid# your X handle"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/icon.jpg?4437d933ce883ba35e5288f3e33907a5"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://naimurashid.github.io/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a href="/" class=""> <img src="/assets/img/RashidLabMainLogo2400x1800.jpeg" height="45" alt=""> </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">about <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/team.html">team </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title"> </h1> <p class="desc"></p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/prof-480.webp 480w,/assets/img/prof-800.webp 800w,/assets/img/prof-1400.webp 1400w," sizes="(min-width: 930px) 270.0px, (min-width: 576px) 30vw, 95vw" type="image/webp"> <img src="/assets/img/prof.jpg?ecd96cc1b91781e4c15972bb37342870" class="img-fluid z-depth-1 rounded" width="100%" height="auto" alt="prof.jpg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> <div class="address"> <p> <span class="name">Naim U. Rashid, PhD</span> <span class="data"> <br>Associate Professor <br>Department of Biostatistics<br>Gillings School of Global Public Health,<br> Lineberger Comprehensive Cancer Center </span> </p> <p class="post">Lineberger 20-020 <br> 450 West Drive <br> University of North Carolina at Chapel Hill <br>Chapel Hill, NC, 27599 </p> </div> </div> <div class="clearfix"> <div class="about-logo"> <img style="padding-bottom: 1rem; margin-right: 2rem" src="/assets/img/RashidLabMainLogo2400x1800.jpeg" height="150"> </div> <p><strong>Overview</strong></p> <p>Dr. Rashid is an <a href="https://sph.unc.edu/adv_profile/naim-rashid-phd/" rel="external nofollow noopener" target="_blank">associate professor</a> (with tenure) in the <a href="https://sph.unc.edu/bios/biostatistics/" rel="external nofollow noopener" target="_blank">Department of Biostatistics</a> at the <a href="https://sph.unc.edu/" rel="external nofollow noopener" target="_blank">UNC Gillings School of Global Public Health</a>, and has a joint appointment at the <a href="https://unclineberger.org/" rel="external nofollow noopener" target="_blank">Lineberger Comprehensive Cancer Center (LCCC)</a>. He currently serves as the Associate Director of the <a href="https://unclineberger.org/biostats/" rel="external nofollow noopener" target="_blank">Lineberger Biostatistics Shared Resource</a>, and co-directs the Biostatistics Cores of the UNC Pancreatic and Breast Cancer SPOREs.</p> <p>Dr. Rashid’s lab is devoted to research addressing basic science, translational, and clinical problems in cancer. His group is deeply involved in the development of novel statistical methodology supporting this aim, with a specific focus in the areas of genomics, precision medicine, clinical trials, and machine learning with applications to pancreatic and breast cancers.</p> <p>Ultimately, the Rashid lab hopes to impact public health by pushing the boundaries of cancer research through innovative new tools, and aid in the development of new interventions and treatments for cancer. Through teaching and mentoring, Dr. Rashid hopes to pass on these passions to his students and help prepare the next generation of statistical researchers.</p> <p><strong>Free Statistical Consultation for Lineberger Members</strong></p> <p>For general inquiries, please email LCCC_BIOS@med.unc.edu to make an appointment with one of our statisticians. We also provide a virtual drop-in clinic on Wednesdays from 12 to 1 PM. To join, please click on the following: <a href="https://zoom.us/j/98595369470" rel="external nofollow noopener" target="_blank">Zoom Drop-In Clinic link</a>. More info on the Lineberger Biostatistics Shared Resource <a href="https://unclineberger.org/biostats/" rel="external nofollow noopener" target="_blank">can be found here</a>.</p> </div> <h2> <a href="/news/" style="color: inherit">news</a> </h2> <div class="news"> <div class="table-responsive"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Aug 15, 2024</th> <td> The updated Rashid lab website has gone live! </td> </tr> </table> </div> </div> <h2> <a href="/publications/" style="color: inherit">selected publications</a> </h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">Biometrics</abbr> </div> <div id="heiling2023efficient" class="col-sm-8"> <div class="title">Efficient computation of high-dimensional penalized generalized linear mixed models by latent factor modeling of the random effects</div> <div class="author"> Hillary M Heiling<sup>*</sup>, Naim U Rashid, Quefeng Li, Xianlu L Peng, Jen Jen Yeh, and Joseph G Ibrahim </div> <div class="periodical"> <em>Biometrics</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://doi.org/10.1093/biomtc/ujae016" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Modern biomedical datasets are increasingly high-dimensional and exhibit complex correlation structures. Generalized linear mixed models (GLMMs) have long been employed to account for such dependencies. However, proper specification of the fixed and random effects in GLMMs is increasingly difficult in high dimensions, and computational complexity grows with increasing dimension of the random effects. We present a novel reformulation of the GLMM using a factor model decomposition of the random effects, enabling scalable computation of GLMMs in high dimensions by reducing the latent space from a large number of random effects to a smaller set of latent factors. We also extend our prior work to estimate model parameters using a modified Monte Carlo Expectation Conditional Minimization algorithm, allowing us to perform variable selection on both the fixed and random effects simultaneously. We show through simulation that through this factor model decomposition, our method can fit high-dimensional penalized GLMMs faster than comparable methods and more easily scale to larger dimensions not previously seen in existing approaches.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">heiling2023efficient</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1093/biomtc/ujae016}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1093/biomtc/ujae016}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Efficient computation of high-dimensional penalized generalized linear mixed models by latent factor modeling of the random effects}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Heiling, Hillary M and Rashid, Naim U and Li, Quefeng and Peng, Xianlu L and Yeh, Jen Jen and Ibrahim, Joseph G}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Biometrics}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{80}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{ujae016}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Oxford University Press}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100">JCO</abbr> </div> <div id="innocenti2024dna" class="col-sm-8"> <div class="title">DNA mutational profiling in patients with colorectal cancer treated with standard of care reveals differences in outcome and racial distribution of mutations</div> <div class="author"> Federico Innocenti, Wancen Mu, Xueping Qu, Fang-Shu Ou, Omar Kabbarah, Charles David Blanke, Alan P Venook, Heinz-Josef Lenz, and Naim U Rashid </div> <div class="periodical"> <em>Journal of Clinical Oncology</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="https://doi.org/10.1200/JCO.23.00825" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>PURPOSECALGB (Alliance)/SWOG 80405 was a randomized phase III trial that in first-line patients with metastatic colorectal cancer (mCRC) treated with bevacizumab or cetuximab with chemotherapy. We aimed to discover novel mutated genes associated with prognosis and differential response to therapy with the biologics.METHODSPrimary tumor DNA from 548 patients was sequenced using FoundationOne. The effect of mutated genes and mutations on overall survival (OS) was tested adjusting for microsatellite instability status, BRAF V600E, all RAS mutations, arm, sex, and age.RESULTSThe median number (lower-upper quartile) of mutated genes was 5 (3-7), 5 (3-6) in microsatellite stable and 12.5 (4.5-32) in microsatellite instability-high tumors. Mutated KRAS and APC were more frequent in Black (53% and 85%) than White (27% and 65%, respectively) patients while BRAF V600E was less frequent in Black (5%) than White (14%) patients. The median OS in patients with BRAF non-V600E (2.2% of patients) was 31.9 months (95% CI, 15.1 to not applicable [NA]) similar to that of BRAF wild-type (WT) patients (31.2 months [95% CI, 29.0 to 33.9]). Mutated LRP1B (10.7% of patients) was associated with improved OS compared with WT LRP1B (hazard ratio, 0.57 [95% CI, 0.40 to 0.80]). RNF43 (5.6% of patients) interacted with treatment arms as, in the cetuximab arm, patients with mutated RNF43 had a median OS of 11.5 (95% CI, 10.8 to NA) months compared with 30.1 (95% CI, 24.9 to 35.3) months in patients with WT RNF43, whereas in the bevacizumab arm, patients with mutated RNF43 had a median OS of 25.0 (95% CI, 14.2 to NA) months compared with 31.3 (95% CI, 29.0 to 34.3) months in patients with WT RNF43.CONCLUSIONThese results can provide new tools to predict patient outcome and improve therapeutic decisions and trial participation in patient minorities. The molecular alterations identified in this study may direct biomarker-driven studies.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">innocenti2024dna</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{DNA mutational profiling in patients with colorectal cancer treated with standard of care reveals differences in outcome and racial distribution of mutations}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Innocenti, Federico and Mu, Wancen and Qu, Xueping and Ou, Fang-Shu and Kabbarah, Omar and Blanke, Charles David and Venook, Alan P and Lenz, Heinz-Josef and Rashid, Naim U}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Journal of Clinical Oncology}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{42}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{4}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{399--409}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Wolters Kluwer Health}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1200/JCO.23.00825}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://doi.org/10.1200/JCO.23.00825}</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="van2020differential" class="col-sm-8"> <div class="title">Differential Transcript Usage Analysis Incorporating Quantification Uncertainty Via Compositional Measurement Error Regression Modeling</div> <div class="author"> Amber Marie Young<sup>*</sup>, S. Van Buren<sup>*</sup>, and Naim U Rashid </div> <div class="periodical"> <em>Biostatistics</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="lim2022" class="col-sm-8"> <div class="title">Deeply-Learned Generalized Linear Models with Missing Data</div> <div class="author"> David K Lim<sup>*</sup>, Naim U Rashid, Junier B Oliva, and Joseph G Ibrahim </div> <div class="periodical"> <em>Journal of Computational and Graphical Statistics</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="rashid2019modeling" class="col-sm-8"> <div class="title">Modeling Between-Study Heterogeneity for Improved Reproducibility in Gene Signature Selection and Clinical Prediction</div> <div class="author"> NU Rashid, Quefeng Li, Jen Jen Yeh, and Joseph G Ibrahim </div> <div class="periodical"> <em>Journal of the American Statistical Association</em>, 2020 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="rashid2019high" class="col-sm-8"> <div class="title">High-Dimensional Precision Medicine From Patient-Derived Xenografts</div> <div class="author"> NU Rashid, Daniel J Luckett, Jingxiang Chen, Michael T Lawson, Longshaokan Wang, Yunshu Zhang, Eric B Laber, Yufeng Liu, Jen Jen Yeh, Donglin Zeng, and  others </div> <div class="periodical"> <em>Journal of the American Statistical Association</em>, 2020 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="rashid2020purity" class="col-sm-8"> <div class="title">Purity Independent Subtyping of Tumors (PurIST), A Clinically Robust, Single-sample Classifier for Tumor Subtyping in Pancreatic Cancer</div> <div class="author"> NU Rashid, Xianlu L Peng, Chong Jin, Richard A Moffitt, Keith E Volmar, Brian A Belt, Roheena Z Panni, Timothy M Nywening, Silvia G Herrera, Kristin J Moore, and  others </div> <div class="periodical"> <em>Clinical Cancer Research</em>, 2020 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="moffitt2015virtual" class="col-sm-8"> <div class="title">Virtual microdissection identifies distinct tumor-and stroma-specific subtypes of pancreatic ductal adenocarcinoma</div> <div class="author"> Richard A Moffitt, Raoud Marayati, Elizabeth L Flate, Keith E Volmar, S Gabriela Herrera Loeza, Katherine A Hoadley, Naim U Rashid, Lindsay A Williams, Samuel C Eaton, Alexander H Chung, and  others </div> <div class="periodical"> <em>Nature genetics</em>, 2015 </div> <div class="periodical"> </div> <div class="links"> </div> </div> </div> </li> </ol> </div> <div class="social"> <div class="contact-icons"> <a href="mailto:%6E%61%69%6D@%75%6E%63.%65%64%75" title="email"><i class="fa-solid fa-envelope"></i></a> <a href="https://scholar.google.com/citations?user=3Cz_lcEAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a> <a href="https://github.com/naimurashid" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-github"></i></a> <a href="https://www.linkedin.com/in/naim-rashid-a767aba" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-linkedin"></i></a> <a href="https://twitter.com/naimurashid#%20your%20X%20handle" title="X" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-x-twitter"></i></a> <a href="https://sph.unc.edu/adv_profile/naim-rashid-phd/" title="Work" rel="external nofollow noopener" target="_blank"><i class="fa-solid fa-briefcase"></i></a> <a href="https://bsky.app/profile/naimurashid.bsky.social" title="Bluesky" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-bluesky"></i></a> </div> <div class="contact-note"></div> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2024 Naim Rashid. #Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. #Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. #Photos from <a href="https://unsplash.com" target="_blank" rel="external nofollow noopener">Unsplash</a>. Last updated: September 17, 2024. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-8PZVKQGFGK"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-8PZVKQGFGK");</script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> </body> </html>